FHEM/14_SD_UT.pm

```diff
@@ -771,6 +845,33 @@ sub SD_UT_Set($$$@) {
 				my $invert = $save;
 				$invert =~ tr/01/10/;									# invert byte 4 (byte5 = inverted byte 4)
 				$msg .= $save.$invert.$msgEnd;
+			############ Medion OR28V ############
+			} elsif ($model eq "OR28V") {
+				my @definition = split(" ", $hash->{DEF});						# split adress from def
+				my $adr = $definition[1] - 1;													# argument 1 - adress
+				my $checksum = ($adr * 16 + oct("0b$save") + 85) & 0x7f;
+				my $laststate = ReadingsVal($name, "state", "fehlt");
+				my $lastbit0 = ReadingsVal($name, "bit0", "fehlt");
+				Log3 $name, 5, "$ioname: SD_UT_Set $name ReadingsVal lastbit0=$lastbit0";
+				my $bit0 = "0";													# bit 0 first press a new button
+				if ($cmd eq $laststate && $cmd ne "power") {
+					$bit0 = $lastbit0 eq "0" ? "1" : "0";	# toggle bit 0 when press the same button again
+				}
+				$bit0 = "1" if ($cmd eq "power");				# bit 0 always 1 by button power
+				Log3 $name, 5, "$ioname: SD_UT_Set $name ReadingsVal laststate=$laststate cmd=$cmd bit0=$bit0";
+				$msg .= $bit0;													# bit 0
+				$msg .= sprintf ("%07b",$checksum);			# bit 1-7
+				if (oct("0b$save") > 42) {	# button
+					$msg .= $bit0;												# bit 8 = bit 0
+				} else {
+					$msg .= $bit0 eq "0" ? "1" : "0";			# bit 8 = bit 0 inverted
+				}
+				$msg .= "0";														# bit 9 always zero
+				$msg .= $save;													# bit 10-15
+				$msg .= sprintf ("%04b",$adr);					# bit 16-19
```

Comment:
Da Du ohnehin schon sprintf verwendest, könntest Du das anhängen auch in die sprintf funktion verlagern.

`$msg .= sprintf ("%s$s%04b%s","0",$save,$adr,$msgEnd);`

Am Resultat verändert das aber nichts.

Reply:
Ich werde es gern deinem Vorschlag anpassen morgen.

Reply:
Habe noch einen Tippfehler korrigiert :)

Reply:
Bei dieser Variante sind aber meine Kommentare weg - also bitte so lassen. Übersichtlicher ist es obendrein.

Reply:
okay die Kommentare hatte ich übersehen :)

Reply:
Hier sollte ja ein Haken dran sein an der Frage :-)
Eine Anpassung erfolgt hier nicht.